src/components/NewOrder.js

```diff
@@ -23,20 +26,33 @@ class NewOrder extends Component {
     this.props.loadItems();
   }
 
-  addOrderItem() {
-    const defaultValue = this.props.masterItems[0];
-    const items = this.state.items.concat(this.state.selectedItem || defaultValue);
+  addEntry() {
+    const selectedItem = Object.assign({}, this.state.selectedItem || this.props.masterItems[0]);
+    selectedItem.comment = this.state.comment;
+    const entries = this.state.entries.concat(selectedItem);
+    this.clearCommentField();
 
-    this.setState({ items });
+    this.setState({ entries });
+  }
+
+  clearCommentField() {
```

Comment:
I think we should use `setState` vs. mutating the `state` object directly. Otherwise, looks like the right way to clear input field.